<launch>
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find dh_hand_driver)/rosconsole.conf"/>
  <arg name="gripper"/>

  <!-- Parameters -->
  <rosparam command="load" file="$(find dh_hand_driver)/config/hardware.yaml"/>
  <rosparam command="load" file="$(find dh_hand_driver)/config/controllers.yaml"/>

  <!-- Driver -->
  <node name="dh_driver" pkg="dh_hand_driver" type="dh_hardware_interface" output="screen">
    <rosparam command="load" ns="usb" file="$(find dh_hand_driver)/config/usb.yaml"/>
    <rosparam command="load" ns="tcp" file="$(find dh_hand_driver)/config/tcp.yaml"/>
  </node>

  <!-- Controller -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" output="screen"
    args="/AG95/controller/state
          /AG95/controller/position/gripper_joint"/>

</launch>
